app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryListFragment.java

```diff
@@ -0,0 +1,268 @@
+        params.setMargins(toPixel(0), toPixel(0), toPixel(23), toPixel(10));
+
+        Button addWordButton = new Button(new ContextThemeWrapper(requireContext(), R.style.User_Dictionary_Button), null, 0);
+        addWordButton.setText(R.string.user_dict_add_word_button);
+        addWordButton.setTextColor(getResources().getColor(android.R.color.white));
+        addWordButton.setPadding(toPixel(30), toPixel(10), toPixel(10), toPixel(10));
+        addWordButton.setCompoundDrawablesWithIntrinsicBounds(null, null, drawable, null);
+        addWordButton.setCompoundDrawablePadding(toPixel(20));
+        addWordButton.setLayoutParams(params);
+        addWordButton.setOnClickListener(v1 -> showAddWordDialog());
+
+        view.addView(addWordButton);
+
+        return view;
+    }
+
+    // TODO : Implement the import/export function in these menus
+/*    @Override
+    public void onCreateOptionsMenu(Menu menu, @NonNull MenuInflater inflater) {
+        *//*menu.add(0, OPTIONS_MENU_EXPORT, 0, R.string.button_backup);
+        menu.add(0, OPTIONS_MENU_IMPORT, 0, R.string.button_restore);*//*
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == OPTIONS_MENU_EXPORT) {
+            return true;
+        }
+        if (item.getItemId() == OPTIONS_MENU_IMPORT) {
+            return true;
+        }
+
+        return false;
+    }*/
+
+    public static TreeSet<String> getUserDictionaryLocalesSet(final Activity activity) {
+        final Cursor cursor = activity.getContentResolver().query(UserDictionary.Words.CONTENT_URI,
+                new String[] { UserDictionary.Words.LOCALE },
+                null, null, null);
+        final TreeSet<String> localeSet = new TreeSet<>();
+
+        if (null == cursor) {
+            // The user dictionary service is not present or disabled. Return null.
+            return null;
+        }
+
+        try {
+            if (cursor.moveToFirst()) {
+                final int columnIndex = cursor.getColumnIndex(UserDictionary.Words.LOCALE);
+                do {
+                    final String locale = cursor.getString(columnIndex);
+                    localeSet.add(null != locale ? locale : "");
+                } while (cursor.moveToNext());
+            }
+        } finally {
+            cursor.close();
+        }
+
+        final InputMethodManager imm = (InputMethodManager)activity.getSystemService(Context.INPUT_METHOD_SERVICE);
+        final List<InputMethodInfo> imis = imm.getEnabledInputMethodList();
+
+        for (final InputMethodInfo imi : imis) {
+            final List<InputMethodSubtype> subtypes = imm.getEnabledInputMethodSubtypeList(imi, true);
+            for (InputMethodSubtype subtype : subtypes) {
+                final String locale = subtype.getLocale();
+                if (!TextUtils.isEmpty(locale)) {
+                    localeSet.add(locale);
+                }
+            }
+        }
```

Comment:
Adding locales from enabled subtypes of other keyboards is a bit misleading considering the function name.
If you think the locales should be added, I suggest clarifying this in the function name, or move it to a separate function.

Reply:
Indeed, there's no need to add that. Thank you very much for pointing it out.
Done with commit e9784f7.